data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java

```diff
@@ -48,12 +49,20 @@ public LayersLocalCacheData(ExternalDirProvider externalDirProvider,
 
     @Override
     public boolean isCached(VectorLayer vectorLayer) {
-        return getVectorLayerFile(vectorLayer).exists();
+        File[] files = getSimilarCacheFiles(vectorLayer);
+
+        return files.length > 0;
     }
 
     @Override
     public URI getCachedURI(VectorLayer vectorLayer) {
-        return getVectorLayerFile(vectorLayer).toURI();
+        File[] files = getSimilarCacheFiles(vectorLayer);
+
+        if (files.length <= 0) {
```

Comment:
I think you should also check whether the length is > 1 and prompt an error if so, because you might encounter the situation of 2 same VLs files only with different severity...